pkg/ui/workspaces/db-console/src/redux/alerts.ts

```diff
@@ -562,6 +566,51 @@ export const clusterPreserveDowngradeOptionOvertimeSelector = createSelector(
   },
 );
 
+////////////////////////////////////////
+// Upgrade not finalized.
+////////////////////////////////////////
+export const upgradeNotFinalizedDismissedSetting = new LocalSetting(
+  "upgrade_not_finalized_dismissed",
+  localSettingsSelector,
+  false,
+);
+
+/**
+ * Warning when all the nodes are running on the new version, but the cluster is not finalized.
+ */
+export const upgradeNotFinalizedWarningSelector = createSelector(
+  numNodesByVersionsSelector,
+  selectClusterSettingVersion,
+  upgradeNotFinalizedDismissedSetting.selector,
+  (versionsMap, clusterVersion, upgradeNotFinalizedDismissed): Alert => {
+    if (upgradeNotFinalizedDismissed) {
+      return undefined;
+    }
+    // Don't show this warning if nodes are on different versions, since there is
+    // already an alert for that (staggeredVersionWarningSelector).
+    if (!versionsMap || versionsMap.size !== 1 || !clusterVersion) {
```

Comment:
could we also skip showing this warning if `cluster.preserve_downgrade_option` is set? if it's set, it is expected that the upgrade should not finalize, and there already is a separate alert if someone forgets to unset `cluster.preserve_downgrade_option`:

https://github.com/cockroachdb/cockroach/blob/35d6537f89ee95ff17b0a2756fbf705c7a75c5b5/pkg/ui/workspaces/db-console/src/redux/alerts.ts#L528-L563

Reply:
Done